File: features/document/src/main/java/com/nivasafinance/features/document/exception/S3ExpirationException.java
package com.nivasafinance.features.document.exception;

import org.springframework.context.MessageSource;

import java.io.Serial;

public class S3ExpirationException extends S3OperationException {
    
    @Serial
    private static final long serialVersionUID = 1L;
    
    public static final String S3_EXPIRATION_INVALID_KEY = "error.document.s3.expiration.invalid";
    
    public S3ExpirationException(String message) {
        super(message);
    }
    
    public S3ExpirationException(MessageSource messageSource) {
        super(S3_EXPIRATION_INVALID_KEY, messageSource);
    }
}

File: features/document/src/main/java/com/nivasafinance/features/document/exception/S3ExpirationException.kt
package com.nivasafinance.features.document.exception

import com.nivasafinance.common.exception.BadRequestException
import org.springframework.context.MessageSource
import org.springframework.context.i18n.LocaleContextHolder
import java.util.Locale

class S3ExpirationException(
    message: String
) : BadRequestException(message) {

    companion object {
        private val locale: Locale = LocaleContextHolder.getLocale()
        const val S3_EXPIRATION_INVALID_KEY = "error.document.s3.expiration.invalid"
    }

    constructor(messageSource: MessageSource) : this(
        messageSource.getMessage(S3_EXPIRATION_INVALID_KEY, arrayOf(""), locale)
    )
}
